packages/components/src/Chips/InternalChipDismissible/InternalChipDismissible.tsx

```diff
@@ -17,8 +17,33 @@ export function InternalChipDismissible(props: InternalChipDismissibleProps) {
     handleChipRemove,
     handleCustomAdd,
     handleWrapperKeyDown,
+    setChipRef,
   } = useInternalChipDismissible(props);
 
+  const chipElementsRef = useRef<Map<string, HTMLElement | null>>(new Map());
+
+  // Effect to update refs when chips are rendered
+  useEffect(() => {
+    // Wait for the DOM to be updated
+    setTimeout(() => {
+      if (wrapperRef.current) {
+        const chipElements =
+          wrapperRef.current.querySelectorAll('[role="option"]');
```

Comment:
fragile but maybe acceptable....

an alternative method could be using React.Children iteration methods and checking the `type` of the `children` rather than mixing in DOM querying